worker/ocspStatus/ocspStatus.go

```diff
@@ -0,0 +1,167 @@
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s decoding issuer for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+	x509Issuer, err := x509.ParseCertificate(x509IssuerCertEncoded)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s parsing issuer for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// grab OCSP response
+	opts := &ocsp.RequestOptions{
+		Hash: crypto.SHA256,
+	}
+	req, err := ocsp.CreateRequest(x509Cert, x509Issuer, opts)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s creating OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+	httpResponse, err := http.Post(x509Cert.OCSPServer[0], "application/ocsp-request", bytes.NewReader(req))
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s making OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// parse response
+	output, err := ioutil.ReadAll(httpResponse.Body)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s reading OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+	resp, err := ocsp.ParseResponse(output, x509Issuer)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s parsing OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	status := Result{
+		Status: resp.Status,
+	}
+	if resp.Status == ocsp.Revoked {
+		status.RevokedAt = resp.RevokedAt
+	}
+
+	out, err := json.Marshal(status)
+	if err != nil {
+		res.Success = false
+		res.Errors = append(res.Errors, err.Error())
+	} else {
+		res.Success = true
+		res.Result = out
+	}
+	resChan <- res
+}
+
+func (ocspStatus) AnalysisPrinter(input []byte, printAll interface{}) (results []string, err error) {
+	var result Result
+	if err = json.Unmarshal(input, &result); err != nil {
+		return nil, fmt.Errorf("ocspStatus Worker: failed to parse results: err=%v", err)
+	}
+
+	results = []string{"* OCSP Status: "}
+	switch result.Status {
+	case ocsp.Good:
+		results = append(results, fmt.Sprintf(" - Not revoked"))
+	case ocsp.Revoked:
+		results = append(results, fmt.Sprintf(" - Revoked at %s\n", result.RevokedAt.Format(time.RFC3339)))
+	default:
+		results = append(results, fmt.Sprintf(" - Unknown status code %d\n", result.Status))
+	}
```

Comment:
Lets put everything on a single line like we did for the CAA and CRL workers. The output should be `fmt.Sprintf("* OCSP: <status>")`. Also, don't include a newline, the printer does that automatically.

Reply:
@jvehent Done!